core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -292,6 +292,8 @@ DECLARE_KEEP_INIT _start
 		 */
 		mul	r1, r0, r1
 		add	sp, r1, r2
+
+		bl	init_canaries
```

Comment:
With this `init_canaries()` will end up being called once for each CPU. How about moving this call to just after the `set_sp` "call" below?

Reply:
Agreed